src/KDSoapClient/KDSoapClientInterface.cpp

```diff
@@ -142,21 +144,29 @@ QNetworkRequest KDSoapClientInterfacePrivate::prepareRequest(const QString &meth
     return request;
 }
 
-QBuffer *KDSoapClientInterfacePrivate::prepareRequestBuffer(const QString &method, const KDSoapMessage &message, const KDSoapHeaders &headers)
+QBuffer *KDSoapClientInterfacePrivate::prepareRequestBuffer(const QString &method, KDSoapMessage &message, const QString &soapAction, const KDSoapHeaders &headers)
 {
     KDSoapMessageWriter msgWriter;
     msgWriter.setMessageNamespace(m_messageNamespace);
     msgWriter.setVersion(m_version);
-    const QByteArray data = msgWriter.messageToXml(message, (m_style == KDSoapClientInterface::RPCStyle) ? method : QString(), headers, m_persistentHeaders, m_authentication);
+    if (m_useWsAddressing) {
+        KDSoapMessageAddressingProperties prop;
+        prop.setAction(soapAction);
+        message.setMessageAddressingProperties(prop);
```

Comment:
This overwrites any existing AddressingProperties. I think you should get messageAddressingProperties before this.

Reply:
Fixed